src/tracedAtomic.ml

```diff
@@ -304,31 +355,45 @@ let check f =
   let tracing_at_start = !tracing in
   tracing := false;
   if not (f ()) then (
-    Printf.printf "Found assertion violation at run %d:\n" !num_runs;
-    List.iter
-      (fun s ->
-        match s with
-        | last_run_proc, last_run_op, last_run_ptr ->
-            let last_run_ptr =
-              Option.map string_of_int last_run_ptr |> Option.value ~default:""
-            in
-            Printf.printf "Process %d: %s %s\n" last_run_proc
-              (atomic_op_str last_run_op)
-              last_run_ptr)
-      !schedule_for_checks;
+    Printf.printf "Found assertion violation at run %d:\n" !num_interleavings;
+    print_execution_sequence stdout;
     assert false);
   tracing := tracing_at_start
 
 let reset_state () =
   finished_processes := 0;
   atomics_counter := 1;
-  num_runs := 0;
+  num_states := 0;
+  num_interleavings := 0;
   schedule_for_checks := [];
+  Trace_tracker.clear_traces ();
   CCVector.clear processes
 
-let trace func =
+let dscheck_trace_file_env = Sys.getenv_opt "dscheck_trace_file"
+
+let dpor func =
   reset_state ();
   let empty_state = do_run func [ (0, Start, None) ] :: [] in
   let empty_clock = IntMap.empty in
   let empty_last_access = IntMap.empty in
   explore func empty_state empty_clock empty_last_access
+
+let trace ?interleavings ?(record_traces = false) func =
+  record_traces_flag := record_traces || Option.is_some dscheck_trace_file_env;
+  interleavings_chan := interleavings;
+
+  dpor func;
+
+  (* print reports *)
+  (match !interleavings_chan with
+  | None -> ()
+  | Some chan ->
+      Printf.fprintf chan "\nexplored %d maximal interleavings and %d states\n"
+        !num_interleavings !num_states);
```

Comment:
Removed the word _maximal_, since usually we refer to interleavings meaning maximal interleavings anyway (and there's no need to be extra rigorous here). As for the `state` it's a bit harder - the key here is that dscheck sees a multicore program as a graph, and then state is just a state. Hopefully the new readme will make that clearer.